### PR TITLE
Implement `Display` for`Attributes`

### PR DESCRIPTION
**Pull request by samdealy**
### Motivation
This PR implements `Display` for `Attributes` in order to make attestation errors more readable and useful (see [54](https://github.com/mobilecoinfoundation/attestation/pull/54)). Previously, we implemented `Display` in the `new_types!` macro and relied on the `Debug` trait. The new `Display` implementation prints out each `Attributes` flag in hex. 

### Future Work
- Implement `Display`for other `mc_sgx_core_types`

### CODE REVIEW

**Comment by samdealy** [comment]
Current dependencies on/for this PR:
* main
  * **PR #308** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/308" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/308?utm_source=stack-comment).

**Review by jcape — COMMENTED** [review]
Looks good, but still needs unit tests for the implementation (that will quiet codecov)---a couple exemplar "this bit value should render as this string" tests should be sufficient

File: core/types/src/attributes.rs [hunk 1]
```diff
@@ -35,6 +42,80 @@ impl Attributes {
...
+                write!(f, "{}", name)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+bitflags! {
+    #[derive(Clone, PartialOrd, Copy, Debug, PartialEq, Eq, Hash)]
```
Comment (jcape) [inline core/types/src/attributes.rs:82]:
```suggestion
    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd)]
```

File: core/types/src/attributes.rs [hunk 2]
```diff
@@ -35,6 +42,80 @@ impl Attributes {
...
+        const PROVISION_KEY = SGX_FLAGS_PROVISION_KEY as u64;
+        /// If set, then the enclave has access to EINITTOKEN key
+        const EINITTOKEN_KEY = SGX_FLAGS_EINITTOKEN_KEY as u64;
+        /// If set enclave uses KSS
+        const KSS = SGX_FLAGS_KSS as u64;
+        /// BIT[55-48] will not be checked */
+        const NON_CHECK_BITS = SGX_FLAGS_NON_CHECK_BITS;
+    }
+
+    #[derive(Clone, PartialOrd, Copy, Debug, PartialEq, Eq, Hash)]
```
Comment (jcape) [inline core/types/src/attributes.rs:100]:
```suggestion
    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd)]
```

**Review by NotGyro — APPROVED** [review]
Looks good to me.

File: core/types/src/attributes.rs (hunk 1 above)
Comment (samdealy) [reply core/types/src/attributes.rs:82]: Done.

File: core/types/src/attributes.rs (hunk 2 above)
Comment (samdealy) [reply core/types/src/attributes.rs:100]: Done.

**Review by jcape — APPROVED** [review]
Looks like either codecov or cargo llvm-cov may be having some issues

**Pull request merged**